src/corePlugins.js

```diff
@@ -143,20 +143,33 @@ export let variantPlugins = {
       })
     }
 
-    for (let [variantName, state] of pseudoVariants) {
-      addVariant(`group-${variantName}`, (ctx) => {
-        let result = typeof state === 'function' ? state(ctx) : state
-
-        return result.replace(/&(\S+)/, ':merge(.group)$1 &')
-      })
+    let variants = {
+      group: ({ label }) =>
+        label ? [`:merge(.group\\<${label}\\>)`, ' &'] : [`:merge(.group)`, ' &'],
+      peer: ({ label }) =>
+        label ? [`:merge(.peer\\<${label}\\>)`, ' ~ &'] : [`:merge(.peer)`, ' ~ &'],
     }
 
-    for (let [variantName, state] of pseudoVariants) {
-      addVariant(`peer-${variantName}`, (ctx) => {
-        let result = typeof state === 'function' ? state(ctx) : state
+    for (let [name, fn] of Object.entries(variants)) {
+      matchVariant(
+        name,
+        (ctx = {}) => {
+          let { label, value = '' } = ctx
+          if (label) {
+            log.warn(`labelled-${name}-experimental`, [
+              `The labelled ${name} feature in Tailwind CSS is currently in preview.`,
+              'Preview features are not covered by semver, and may be improved in breaking ways at any time.',
+            ])
+          }
 
-        return result.replace(/&(\S+)/, ':merge(.peer)$1 ~ &')
-      })
+          let result = normalize(typeof value === 'function' ? value(ctx) : value)
```

Comment:
This supports value as a fn but that's not exercised in any of the tests — this a leftover?

Reply:
nevermind. it's for the visited variant.